README.md

```diff
@@ -0,0 +1,60 @@
+# tool-upgrade-go
+
+A library for golang that helps upgrade tools to newer versions with ease.
+
+## Usage
+
+### Repository structure
```

Comment:
As someone who is unfamiliar with this tool, my first time reading through this section was confusing. Am I required to perform the last three commands to have a properly structured repository? Where does the web server or artifact repository come into play? Is this repository structure referring to my code's repo, or the web server/artifact repository?

Reply:
Yes, the first couple commands show artifacts built for each supported architecture and OS using the specific naming convention. The last three commands are an example of what is required to make the artifacts available as a discrete version, and to increment the available version for a release channel, according to the convention that this library uses. The example is for a basic web server which serves as a primitive artifact repository. You can use artifactory or another specific tool instead of a web server, as long as the artifacts are in the conventional locations from the specified repo base.

Perhaps adding something like this to the end of the example would help?

```
% find path/to/coolTool
path/to/coolTool
path/to/coolTool/1.2.2
path/to/coolTool/1.2.2/coolTool_darwin_amd64
path/to/coolTool/1.2.2/coolTool_linux_amd64
path/to/coolTool/1.2.2/coolTool_windows_amd64
path/to/coolTool/1.2.3
path/to/coolTool/1.2.3/coolTool_darwin_amd64
path/to/coolTool/1.2.3/coolTool_linux_amd64
path/to/coolTool/1.2.3/coolTool_windows_amd64
path/to/coolTool/.version-latest
```

Reply:
Sure, that could help. Once I understood that this is a artifact repository, the process makes more sense.